### DISCUSSION TITLE
Subscriptions - Is CardForm the only way to go? Or is Bricks or Checkout Pro available for that too?

### DISCUSSION THREAD

**Discussion by fedefrei** [mercadopago/sdk-js#182] [Q&A] [answered]
I'm implementing the cardform on my website to get the card token Id and make a subscription with that. But I'm wondering if is there another way to do this without the coding of the card form.

Thanks!!

**Comment by wimsantos-mp** [comment, marked as answer]
@fedefrei how are you? I hope you are fine!

Thanks for contact us and give an important contribution.
Let me know something, are you interesting in which type of payment methods? Debit and credit cards only? If it is, we suggest you to use our bricks called **CardPayment**, now, if you need more payment types we recommend our Payment Brick, this one is a selector which has more payment options for your customers.

In this [URL](https://www.mercadopago.com.br/developers/en/docs/checkout-bricks/landing ) you can find how to implement both solutions: 

Huge Thanks!

**Reply by fedefrei** [reply]
@wimsantos-mp I think Debit/Credit cards for now will be enough. They work for subscriptions? I think I only need to get the card token ID and send that when subscribing the user.

However I'll try that out and if it works with subs I'll let you know. Thanks!

**Reply by wimsantos-mp** [reply]
Hi, @fedefrei, according to the specs of subscriptions on devsite, we can't see no reason to not work CardPayment Brick as a frontend layer in your experience. Thank you.

**Reply by fedefrei** [reply]
Just in case anyone is wondering the same. I was able to complete the subscription flow using the bricks. Thanks @wimsantos-mp !!

**Reply by Acruel** [reply]
idk if this conversation still active, but, how can I receive the "Card token" for the suscription, if they pay with the mercadopago wallet?

**Reply by lemazzo** [reply]
Hello, @Acruel!
You do not receive a “card token” for the subscription if the payment was made using the Mercado Pago Wallet. This is because “card token” is only for card payments.
Thanks for your contact!
